src/game/client/draw/drawable.cpp

```diff
@@ -175,3 +212,672 @@ void Drawable::Clear_And_Set_Model_Condition_Flags(
+        s_animationTemplates[5] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[5]);
+        s_animationTemplates[6] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[6]);
+        s_animationTemplates[7] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[7]);
+        s_animationTemplates[8] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[8]);
+        s_animationTemplates[9] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[9]);
+        s_animationTemplates[10] = nullptr;
+        s_animationTemplates[11] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[11]);
+        s_animationTemplates[12] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[12]);
+        s_animationTemplates[13] = g_theAnim2DCollection->Find_Template(s_theDrawableIconNames[13]);
+        s_staticImagesInited = true;
+    }
+}
+
+void Drawable::Kill_Static_Images()
+{
+    // static image instances are destroyed by the ImageCollection destructor and don't need to be freed here.
+    if (s_animationTemplates != nullptr) {
+        delete[] s_animationTemplates;
+        s_animationTemplates = nullptr;
+    }
+}
+
+bool Drawable::Draws_Any_UI_Text()
+{
+    if (!Is_Selected()) {
+        return false;
+    }
+
+    Object *object = Get_Object();
+
+    if (object == nullptr || !object->Is_Locally_Controlled()) {
+        return false;
+    }
+
+    Player *player = object->Get_Controlling_Player();
+    int squad = player->Get_Squad_Number_For_Object(object);
+
+    if (squad > -1 && squad < 10) {
+        return true;
+    }
+
+    m_groupString = nullptr;
+    return object->Get_Formation_ID() != FORMATION_UNK;
+}
+
+bool Compute_Health_Region(Drawable const *drawable, IRegion2D &region)
+{
+    if (drawable == nullptr) {
+        return false;
+    }
+
+    const Object *object = drawable->Get_Object();
+
+    if (object == nullptr) {
+        return false;
+    }
+
+    Coord3D pos;
+    object->Get_Health_Box_Position(pos);
+    ICoord2D screen;
+
+    if (!g_theTacticalView->World_To_Screen_Tri(&pos, &screen)) {
+        return false;
+    }
+
+    float width;
+    float height;
+
+    if (!object->Get_Health_Box_Dimensions(width, height)) {
+        return false;
+    }
+
+    float zoom = 1.0f / g_theTacticalView->Get_Zoom();
+    height = height * zoom;
+    width = 3.0f;
+    region.lo.x = screen.x - height * 0.45f;
+    region.lo.y = screen.y - width * 0.5f;
+    region.hi.x = region.lo.x + height;
+    region.hi.y = region.lo.y + width;
+    return true;
+}
+
+void Drawable::Draw_Icon_UI()
+{
+    if (g_theGameLogic->Get_Draw_Icon_UI() && g_theScriptEngine->Get_Fade() == ScriptEngine::FADE_NONE) {
+        IRegion2D healthregion;
+        IRegion2D *region = nullptr;
+
+        if (Compute_Health_Region(this, healthregion)) {
+            region = &healthregion;
+        }
+
+        Object *object = Get_Object();
+
+        if (object) {
+            Draw_Health_Bar(region);
```

Comment:
Is this proper? If `Compute_Health_Region` would return false, then `region` would be null here.

Reply:
This is correct because functions are doing checks on Region before using it.